src/graph/xml.cc

```diff
@@ -636,15 +636,15 @@ ncclResult_t ncclTopoGetXmlFromGpu(struct ncclXmlNode* pciNode, uint32_t rocmDev
   int sm;
   NCCLCHECK(xmlGetAttrInt(gpuNode, "sm", &sm));
 
-  int gcn;
+  const char* gcn;
   NCCLCHECK(xmlGetAttrIndex(gpuNode, "gcn", &index));
   if (index == -1) {
     hipDeviceProp_t devProp;
     CUDACHECK(hipGetDeviceProperties(&devProp, 0));
-    gcn = devProp.gcnArch;
-    NCCLCHECK(xmlSetAttrInt(gpuNode, "gcn", gcn));
+    gcn = devProp.gcnArchName;
+    NCCLCHECK(xmlSetAttr(gpuNode, "gcn", gcn));
   }
-  NCCLCHECK(xmlGetAttrInt(gpuNode, "gcn", &gcn));
+  NCCLCHECK(xmlGetAttrStr(gpuNode, "gcn", &gcn));
```

Comment:
Changing "gcn" from int to str breaks topo xml loading function, I tested on gfx908 and saw segfault.